### DISCUSSION TITLE
Código dos Municípios/Estados

### DISCUSSION THREAD

**Discussion by vitorsilverio** [BrasilAPI/BrasilAPI#245] [Ideas]
Seria interessante se api de CEP retornasse os códigos do IBGE do Estado e município
[(Código dos Municípios IBGE)](https://www.ibge.gov.br/explica/codigos-dos-municipios.php). Esses códigos são utilizados para na emissão de NF-e por exemplo ajudaria muito em vários sistemas de cadastro se ao coletar o endereço já fosse possível obter esses códigos.

Pelo que entendi os ceps na verdade não são obtidos por uma implementação nesse repositório, mas sim no modulo `cep-promise`, seria melhor implementar algo assim nesse modulo ou nesse repositório?

**Comment by vitorsilverio** [comment]
Desculpa, depois que vi que o cep-promise é um modulo seu e que já havia um issue relacionado a questão. Vou fechar esse issue por ser uma questão duplicada. 
Duplicado de (https://github.com/filipedeschamps/cep-promise/issues/153)

**Comment by filipedeschamps** [comment]
Opa meu caro, olhando aqui e forma superficial, acho que isso ficaria na responsabilidade desse repositório aqui, digo isso porque se for no `cep-promise` precisaremos garantir uma interface igual vindo de todos os providers (que não vão ter).

Então seria ótimo se ao menos o BrasilAPI fosse um dos providers que retornasse isso na API.

**Comment by vitorsilverio** [comment]
@filipedeschamps fiz uma API em java para ter os dados de municipio do IBGE
https://github.com/vitorsilverio/ibge-dtb quando eu achar um local free para hospedar vai ser possivel obter esses dados a partir dai para incluir na BrasilAPI.
A ideia desse projeto é que ele baixa a planilha disponibilizada e transforma nessa API.
```$xslt
GET /api/municipios/search/nome_e_estado?nomeCidade=S%C3%A3o+Paulo&siglaEstado=SP
```
Resultado:
```$xslt
{
  "_embedded" : {
    "municipios" : [ {
      "nome" : "São Paulo",
      "codigoIBGE" : 3550308,
      "_links" : {
        "self" : {
          "href" : "http://localhost:8080/api/municipios/3550308"
        },
        "municipio" : {
          "href" : "http://localhost:8080/api/municipios/3550308"
        },
        "estado" : {
          "href" : "http://localhost:8080/api/municipios/3550308/estado"
        }
      }
    } ]
  },
  "_links" : {
    "self" : {
      "href" : "http://localhost:8080/api/municipios/search/nome_e_estado?nomeCidade=S%C3%A3o+Paulo&siglaEstado=SP"
    }
  }
}
```
fiz essa pesquisa por nome e estado para poder integrar ao CEP

**Comment by vitorsilverio** [comment]
Coloquei a API online no heroku https://ibge-dtb.herokuapp.com/api/

**Comment by lucianopf** [comment]
@vitorsilverio a ideia é bem boa, curti!

Tentei acessar o link mas ele não está disponível 😢 

Qual a complexidade do serviço? Pergunto pq talvez a gnt conseguisse recriar o projeto dentro do BrasilAPI e utilizar as próprias serverless functions da Zeit pra hospedar o serviço 😬

**Comment by vitorsilverio** [comment]
@lucianopf o heroku deve ter ficado fora do ar, mas agora já está funcionando.
O projeto basicamente ta baixando o zip com a planilha extraindo os dados da planilha num banco que roda em memoria. Foi feito em java com spring boot. E atualmente esta hospedado no plano mais simples do heroku

**Comment by ghost** [comment]
O IBGE disponibiliza uma API desse tipo: https://servicodados.ibge.gov.br/api/docs/localidades?versao=1 . Ela não poderia ser útil para resolver esse problema ?
